### PR TITLE
Added github workflows for CI and publishing docker image to replace travis

### PR DESCRIPTION
**Pull request by mraarif**
Relevant JIRA issue here: https://openedx.atlassian.net/browse/BOM-2105

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Python CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (jmbowman) [inline .github/workflows/ci.yml:7]: Repeating from other reviews: Ned has suggested that we run the tests on all PRs, not just the ones targeting master.  Discussion at https://edx-internal.slack.com/archives/CD89W7Z7F/p1606849364465500

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  publish_docker_image:
+    name: Publish Docker Image
+    runs-on: ubuntu-20.04
+    if: github.event_name == 'push'
```
Comment (jmbowman) [inline .github/workflows/docker-publish.yml:12]: With the `on` condition above, would `event_name` ever be something other than "push"?  Seems redundant, but haven't looked into it in depth.

File: requirements/ci.txt [hunk 3]
```diff
@@ -12,7 +12,7 @@ chardet==3.0.4            # via -r requirements/base.txt, requests
 click-log==0.3.2          # via -r requirements/base.txt, edx-lint
```
Comment (jmbowman) [inline requirements/ci.txt:12]: Doesn't the `upgrade` target in `Makefile` need to change to accommodate the rename?

Comment (mraarif) [reply requirements/ci.txt:12]: it did need an update in `Makefile`

**Review by jmbowman — APPROVED** [review]

**Review by jmbowman — APPROVED** [review]

**Pull request merged**